docs/source/developers/guide/step_by_step/set_up.rst

```diff
@@ -27,3 +27,95 @@
 ******
 Set up
 ******
+
+Install and setup Git
+=====================
+
+The Arrow project is developed using `Git <https://git-scm.com/>`_
+for version control which is easily available for all common
+operating systems.
+
+You can follow the instructions to install Git from GitHub
+where Arrow repository is hosted, following
+`the quickstart instructions <https://docs.github.com/en/get-started/quickstart/set-up-git>`_.
+
+When Git is set up do not forget to configure your name and email
+
+.. code:: console
+
+   $ git config --global user.name "Your Name"
+   $ git config --global user.email your.email@example.com
+
+and `authenticate with GitHub <https://docs.github.com/en/get-started/quickstart/set-up-git#next-steps-authenticating-with-github-from-git>`_
+as this will allow you to interact with GitHub without typing
+a username and password each time you execute a git command.
+
+.. note::
+
+   This guide assumes you are comfortable working from the command line.
+   Some IDEs allow you to manage a git repository, but may implicitly run
+   unwanted operations when doing so (such as creating project files).
+
+Get the source code
+===================
+
+Fork the repository
+-------------------
+
+The Arrow GitHub repository contains both the Arrow C++ library plus
+libraries for other languages such as Go, Java, Matlab, Python, R, etc.
+The first step to contributing is to create a fork of the repository
+in your own GitHub account.
+
+1. Go to `<https://github.com/apache/arrow>`_.
+
+2. Press Fork in the top right corner.
+
+   .. figure:: github_fork.jpeg
+      :scale: 50 %
+      :alt: Fork the Apache Arrow repository on GitHub.
+
+      The icon to fork the Apache Arrow repository on GitHub.
+
+3. Choose to fork the repository to your username so the fork will be
+   created at ``https://github.com/<your username>/arrow``.
+
+Clone the repository
+--------------------
+
+Next you need to clone the repository (this is where the command line
+comes in). It is *highly recommended* to clone the repository using
+the command line or a Git client. Cloning it in an IDE might create
+unnecessary headaches - for example, cloning it in RStudio assumes the
+whole repository is an RStudio project and will create a ``.Rproj``
+file in the root directory.
```

Comment:
Hmm, so I notice we warn against IDEs twice (once here and once above). Should we remove one of them?